frontend/src/reducers/user.js

```diff
@@ -0,0 +1,122 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  login: {
+    accessToken: null,
+    userId: 0,
+    secretMessage: null,
+    errorMessage: null,
+  },
+};
+
+export const user = createSlice({
+  name: "user",
+  initialState: initialState,
+  reducers: {
+    setAccessToken: (state, action) => {
+      const { accessToken } = action.payload;
+      state.login.accessToken = accessToken;
+    },
+
+    setUserId: (state, action) => {
+      const { userId } = action.payload;
+      state.login.userId = userId;
+    },
+
+    setFirstName: (state, action) => {
+      const { firstName } = action.payload;
+      state.login.firstName = firstName;
+    },
+
+    setLastName: (state, action) => {
+      const { lastName } = action.payload;
+      state.login.lastName = lastName;
+    },
+
+    setSecretMessage: (state, action) => {
+      const { secretMessage } = action.payload;
+      state.login.secretMessage = secretMessage;
+    },
+
+    setErrorMessage: (state, action) => {
+      const { errorMessage } = action.payload;
+      state.login.errorMessage = errorMessage;
+    }
+  },
+});
+
+// Thunks
+export const login = (email, password) => {
+  const LOGIN_URL = 'https://signinprojecttechnigo.herokuapp.com/sessions';
+  return (dispatch) => {
+    fetch(LOGIN_URL, {
+      method: 'POST',
+      body: JSON.stringify({ email, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((res) => {
+        if (res.ok /* if 200, 201, 204 */) {
+          return res.json();
+        }
+
+        // Not OK
+        throw 'Unable to sign in. Please check thath your email and password are correct';
+      })
+      .then((json) => {
+        // Save the login info
+        dispatch(
+          user.actions.setAccessToken({
+            accessToken: json.accessToken,
+          })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+      })
+      .catch((err) => {
+        dispatch(logout());
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+};
+
+export const getSecretMessage = () => {
+  const USERS_URL = 'https://signinprojecttechnigo.herokuapp.com/users';
+  return (dispatch, getState) => {
+    const accessToken = getState().user.login.accessToken;
+    const userId = getState().user.login.userId;
+    const firstName = getState().user.login.firstName;
+    const lastName = getState().user.login.lastName;
+    // Include userId in the path
+    fetch(`${USERS_URL}/${userId}/users`, {
```

Comment:
Your endpoint looks like this: `/users/:id`, so I think this fetch should be to: `${USERS_URL}/${userId}`.